### ISSUE TITLE
In autobuilder `--debug`, drop into debug shell if products are unsatisfied

### ISSUE THREAD

**Issue by Keno** [JuliaPackaging/BinaryBuilder.jl#1124]
Usually in that case something went wrong, so it would be useful to have a shell to look around at what actually happened.

**Comment by giordano** [comment]
Good idea, but to be clear I don't think we can recreate exactly the same environment as before, for example environment variables set before would have been gone, like anything that had been written to tmpfs
